contracts/evm/contracts/adapters/CentralizedConnection.sol

```diff
@@ -76,28 +76,29 @@ contract CentralizedConnection is Initializable, IConnection {
     ) external payable override {
         require(msg.sender == xCall, "Only Xcall can call sendMessage");
         uint256 fee;
-        if (sn >= 0) {
+        if (sn > 0) {
             fee = this.getFee(to, true);
-        } else {
+        } else if (sn == 0) {
             fee = this.getFee(to, false);
         }
         require(msg.value >= fee, "Fee is not Sufficient");
```

Comment:
Will this not fail on reponse meesages since fee is null?
You could require directly in the if cases or define fee to be 0 by default

Reply:
on getFee method, we have used getOrDefault, so that won't be case I think